src/data-table/data-table.tsx

```diff
@@ -28,27 +32,55 @@ export const DataTable = ({
     loadState = "success",
     enableMultiSelect,
     selectedIds,
+    disabledIds,
     enableSelectAll,
+    enableSelectionBar,
     emptyView,
+    selectionBarContent,
+    enableStickyHeader = false,
     renderCustomEmptyView,
     onHeaderClick,
     onSelect,
     onSelectAll,
+    onClearSelectionClick,
     ...otherProps
 }: DataTableProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const [isAtBottom, setIsAtBottom] = useState(false);
+    const [showLastBorder, setShowLastBorder] = useState(false);
+    const tableRef = useRef<HTMLInputElement>();
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+    useEffect(() => {
+        scrollHandler();
+        checkLastBorder();
+    }, []);
+
     // ===========================================================================
     // HELPER FUNCTIONS
     // ===========================================================================
-    const isAllCheckBoxSelected = (): boolean => {
+    const isAllCheckboxSelected = (): boolean => {
         return selectedIds?.length === rows.length;
     };
 
+    const isIndeterminateCheckbox = (): boolean => {
+        return selectedIds?.length !== 0 && !isAllCheckboxSelected();
```

Comment:
noticed that the indeterminate state was showing up when no selection is passed, just some reworking of the length check

```suggestion
        return selectedIds && selectedIds.length && !isAllCheckboxSelected();
```